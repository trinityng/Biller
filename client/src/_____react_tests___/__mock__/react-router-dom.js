import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;