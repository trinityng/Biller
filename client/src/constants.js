export const HOST = 'http://localhost:8000';

export const USER_EMAIL = 'USER_EMAIL';
export const UPDATE_PROFILE = 'UPDATE_PROFILE';
// User Token
export const USER_TOKEN = 'USER_TOKEN';
export const SET_USER_INFO = 'SET_USER_INFO';

// Server URI
export const LOGIN_URI = '/auth/login/';
export const REGISTER_URI = '/auth/register/';
export const PLAID_ACCESS_TOKEN_URI = '/plaid/access-token/';
export const PLAID_TRANSACTIONS_URI = '/plaid/transactions/';
export const PLAID_TRANSACTIONS_EACH_URI =
  '/plaid/transactions-of-each-account/';
export const PLAID_CATEGORIES_URI = '/plaid/categories/';
export const PLAID_NET_WORTH_URI = '/plaid/net-worth/';
export const PLAID_MONTHLY_EXPENSES_URI = '/plaid/monthly-total-expenses/';
export const PLAID_BILLS_URI = '/plaid/bills/';

// Auth
export const LOGIN_FAILED = 'LOGIN_FAILED';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESSFULLY';

export const REGISTER_FAILED = 'REGISTER_FAILED';
export const REGISTER_SUCCESS = 'REGISTER_SUCCESSFULLY';

export const LOGOUT = 'LOGOUT';

// Plaid
export const PLAID_PRODUCT = ['transactions'];
export const PLAID_PUlLIC_TOKEN = 'PLAID_PUlLIC_TOKEN';
export const PLAID_PUlLIC_KEY = '66974676d9f0b1bcf30d24f66881e0';
export const PLAID_DEV_ENV = 'development';
export const PLAID_SB_ENV = 'sandbox';

export const PLAID_LOGIN_SUCCESS = 'PLAID_LOGIN_SUCCESSFULLY';
export const PLAID_LOGIN_FAILED = 'PLAID_LOGIN_FAILED';

export const PLAID_GET_TRANSACTIONS_SUCCESS = 'PLAID_GET_TRANSACTIONS_SUCCESS';
export const PLAID_GET_TRANSACTIONS_FAILED = 'PLAID_GET_TRANSACTION_FAILED';

export const PLAID_GET_TRANSACTIONS_EACH_SUCCESS =
  'PLAID_GET_TRANSACTIONS_EACH_SUCCESS';
export const PLAID_GET_TRANSACTIONS_EACH_FAILED =
  'PLAID_GET_TRANSACTIONS_EACH_FAILED';

export const PLAID_GET_CATEGORY_SUCCESS = 'PLAID_GET_CATEGORY_SUCCESS';
export const PLAID_GET_CATEGORY_FAILED = 'PLAID_GET_CATEGORY_FAILED';

export const PLAID_GET_NET_WORTH_SUCCESS = 'PLAID_GET_NET_WORTH_SUCCESS';
export const PLAID_GET_NET_WORTH_FAILED = 'PLAID_GET_NET_WORTH_FAILED';

export const PLAID_GET_MONTHLY_EXPENSE_SUCCESS =
  'PLAID_GET_MONTHLY_EXPENSE_SUCCESS';
export const PLAID_GET_MONTHLY_EXPENSE_FAILED =
  'PLAID_GET_MONTHLY_EXPENSE_FAILED';

export const PLAID_GET_ALL_BILLS_SUCCESS = 'PLAID_GET_ALL_BILLS_SUCCESS';
export const PLAID_GET_ALL_BILLS_FAILED = 'PLAID_GET_ALL_BILLS_FAILED';
