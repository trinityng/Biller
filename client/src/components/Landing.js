import React from 'react';
import './Landing.css';
import { Link } from 'react-router-dom';
import { Button, Layout, Menu, Breadcrumb } from 'antd';

const { Header, Footer, Content } = Layout;

const Landing = () => {
  return (
    <div style={{ height: '75vh' }} className="container valign-wrapper">
      <div className="row">
        <div className="col s12 center-align">
          <h4>
            <b>Landing Page </b>
          </h4>
          <br />
          <div className="col s6">
            <Link to="/login">
              <Button type="primary"> Login </Button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Landing;
