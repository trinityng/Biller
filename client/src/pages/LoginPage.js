import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect, withRouter } from 'react-router-dom';

import { login } from '../api/login.api';
import Login from '../components/Login.jsx';
import '../components/static/style.js';

const LoginPage = props => {
  const onClick = values => {
    console.log(values);
    props.login({
      email: values.email,
      password: values.password,
    });
  };

  const user = props.user || {};
  const { access_token } = user;
  console.log('user: ', user);
  console.log('access_token: ', access_token);

  return access_token ? (
    <Redirect to="/" user={user} />
  ) : (
    <Login onClick={e => onClick(e)} />
  );
};

// Store
function mapStateToProps(state) {
  return {
    user: state.user,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({ login }, dispatch);
}

export default connect(
  mapStateToProps,
  matchDispatchToProps
)(withRouter(LoginPage));
