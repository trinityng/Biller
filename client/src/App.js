import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RouterComponent from './router/RouterComponent';
import { setUserToken } from './store/actions/auth.action';
import { USER_EMAIL, USER_TOKEN } from './constants';

class App extends Component {
  componentDidMount() {
    const email = localStorage.getItem(USER_EMAIL);
    const token = localStorage.getItem(USER_TOKEN);
    if (token) {
      this.props.setUserToken({ token });
    }
  }
  render() {
    const user = this.props.user || {};
    return (
      <div>
        <RouterComponent user={user} />
      </div>
    );
  }
}
function mapStateToProps(state) {
  return {
    user: state.user,
  };
}
function matchDispatchToProps(dispatch) {
  return bindActionCreators({ setUserToken }, dispatch);
}
export default connect(mapStateToProps, matchDispatchToProps)(App);
