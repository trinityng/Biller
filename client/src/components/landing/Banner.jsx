import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import QueueAnim from 'rc-queue-anim';
import { Button } from 'antd';
import BannerImage from './BannerImage';
import { USER_TOKEN } from '../../constants';

class Banner extends React.PureComponent {
  static propTypes = {
    className: PropTypes.string,
  };
  static defaultProps = {
    className: 'home-banner',
  };
  render() {
    const { className } = this.props;
    const token = localStorage.getItem(USER_TOKEN);
    return (
      <div className={`home-layout-wrapper ${className}`}>
        <div className="home-layout">
          <QueueAnim
            className={`${className}-content-wrapper`}
            delay={300}
            ease="easeOutQuart"
          >
            <h1 key="h2">Biller</h1>
            <p key="p">
              Your Personal Bill Assistant. <br />
              Join us for free today and never worry about your bill again
            </p>
            <span key="button">
              <Link to="/register">
                <Button
                  type="primary"
                  shape="round"
                  onClick={() => {
                    window.location.href = '/register';
                  }}
                >
                  Sign up
                </Button>
              </Link>
              {token ? (
                <Link to="/login">
                  <Button
                    type="primary"
                    shape="round"
                    onClick={() => {
                      window.location.href = '/dashboard';
                    }}
                  >
                    Dashboard
                  </Button>
                </Link>
              ) : (
                <Link to="/login">
                  <Button
                    type="primary"
                    shape="round"
                    onClick={() => {
                      window.location.href = '/login';
                    }}
                  >
                    Login
                  </Button>
                </Link>
              )}
            </span>
          </QueueAnim>
          <div className={`${className}-image-wrapper`}>
            <BannerImage />
          </div>
        </div>
      </div>
    );
  }
}

export default Banner;
