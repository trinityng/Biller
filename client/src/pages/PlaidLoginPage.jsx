import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Redirect, withRouter } from 'react-router-dom';

import { plaidLogin } from '../api/plaid.api';
import SetupAccount from '../components/setup/SetupAccounts';
import { updateProfile } from '../store/actions/auth.action';
import { USER_TOKEN, ACCOUNTS_INFO } from '../constants';

import './Pages.css';

class PlaidLoginPage extends Component {
  constructor(props) {
    super(props);

    this.onSuccess = this.onSuccess.bind(this);
  }

  componentDidMount() {
    document.body.style.backgroundColor = '#F0F2F5';
  }

  onExit() {}

  onSuccess(public_token, metadata) {
    console.log(public_token);
    const { user, plaid } = this.props;
    const { email, token } = user;
    console.log('[SetupForm] user: ', user);

    this.props.plaidLogin({
      email,
      token,
      public_token,
    });

    const previousForms = JSON.parse(localStorage.getItem(ACCOUNTS_INFO)) || [];
    const forms = [
      ...previousForms,
      {
        accountNum: metadata.accounts.length,
        bankName: metadata.institution.name,
      },
    ];

    console.log('plaid.errors: ' + plaid.errors);
    if (!plaid.errors) {
      this.props.updateProfile({ has_profile: true });
      localStorage.setItem(ACCOUNTS_INFO, JSON.stringify(forms));
    }
  }

  render() {
    const token = localStorage.getItem(USER_TOKEN);
    const { user } = this.props;
    const forms = JSON.parse(localStorage.getItem(ACCOUNTS_INFO)) || [];

    console.log('forms: ', forms);
    return token ? (
      <SetupAccount
        forms={forms}
        has_profile={user.has_profile}
        onSuccess={this.onSuccess}
      />
    ) : (
      <Redirect to="/login" />
    );
  }
}

function mapStateToProps(state) {
  const { user, plaid } = state;
  return {
    user,
    plaid,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({ updateProfile, plaidLogin }, dispatch);
}

export default connect(
  mapStateToProps,
  matchDispatchToProps
)(withRouter(PlaidLoginPage));
