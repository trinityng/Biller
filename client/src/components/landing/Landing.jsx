import React from 'react';
import DocumentTitle from 'react-document-title';
import Header from './Header';
import Banner from './Banner';
import Page1 from './Page1';
import Page2 from './Page3';
import Page3 from './Page2';
import Footer from './Footer';

import '../static/style.js';

class Landing extends React.PureComponent {
  render() {
    return (
      <div className="home-page">
        <Header key="header" />
        <Banner key="banner" />
        <Page1 key="page1" />
        <Page3 key="page3" />
        <Page2 key="page2" />  
        <Footer key="footer" />
        <DocumentTitle title="Biller" />
      </div>
    );
  }
}
export default Landing;
