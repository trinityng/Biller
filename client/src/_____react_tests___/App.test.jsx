import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import GoogleAnalytics from 'react-ga';

import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import Store from '../store/store';
import storageMock from './__mock__/MockLocalStore'
import { mount, configure } from 'enzyme';

import App from '../App';
import Landing from '../components/landing/Landing';
import * as consts from '../constants.js'
import LoginPage from '../pages/LoginPage';
import history from '../router/History'
import RegisterPage from '../pages/RegisterPage';
import BillOverview from '../views/BillOverView/BillOverview';
import PlaidLoginPage from '../pages/PlaidLoginPage';
import Banner from '../components/landing/Banner';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const mockData = {
  data: "mock",
  email: "mock@mock.com",
  token: "mock_token",
  first_name: "mock_first",
  last_name: "mock_last"
}

const mockPayload = {
  data: "mock",
  email: "mock@mock.com",
  token: "mock_token",
  first_name: "mock_first",
  last_name: "mock_last",
  graph_data: [ ],
  net_worth: 200.00,
  category_expense: [ ],
  monthly_expenses: [ ],
  monthly_income: [ ],
}

configure({adapter: new Adapter()});

describe('main App component', () => {
  const mockAxios = new MockAdapter(axios);
  mockAxios.onPost(`${consts.HOST}${consts.PLAID_ACCESS_TOKEN_URI}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_TRANSACTIONS_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_TRANSACTIONS_EACH_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_CATEGORIES_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_NET_WORTH_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_MONTHLY_EXPENSES_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_MONTHLY_INCOME_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_BILLS_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_GRAPH_DATA_URI}?email=${mockData.email}`, mockData).reply(200, mockPayload);
  mockAxios.onGet(`${consts.HOST}${consts.PLAID_GET_NOTIFICATIONS_URI}?email=${mockData.email}`,mockData).reply(200, mockPayload);
  mockAxios.onPost(`${consts.HOST}${consts.PLAID_MARK_NOTIFICATION_READ_URI}`, mockData).reply(200, mockPayload);

  let wrapper;

  beforeEach(() => {
    GoogleAnalytics.set({});
    window.localStorage = storageMock();
    wrapper = mount(
      <Provider store={Store}>
        <App />
      </Provider>
    );
    history.replace("/");
  });

  it('should render correctly as connected component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render landing page', () => {
    history.push("/");
    wrapper.update();
    expect(wrapper.find(Landing).length).toEqual(1);
  });

  describe('when user has no token', () => {
    it('should go to login page', () => {
      history.push('/login');
      wrapper.update();
      expect(wrapper.find(LoginPage).length).toEqual(1);
    });

    it('should go to register page', () => {
      history.push('/register');
      wrapper.update();
      expect(wrapper.find(RegisterPage).length).toEqual(1);
    })

    it('should not go to setup page', () => {
      history.push('/setup');
      wrapper.update();
      expect(wrapper.find(PlaidLoginPage).length).toEqual(0);
    })

    it('should not go to on dashboard', () => {
      history.push('/dashboard');
      wrapper.update();
      expect(wrapper.find(BillOverview).length).toEqual(0);
    })
  })

  describe('when user has token', () => {
    beforeEach(() => {
      localStorage.setItem(consts.USER_TOKEN, mockData.token);
      localStorage.setItem(consts.USER_EMAIL, mockData.email);
      localStorage.setItem(consts.USER_FIRST_NAME, mockData.first_name);
      localStorage.setItem(consts.USER_LAST_NAME, mockData.last_name);
      wrapper = mount(
        <Provider store={Store}>
          <App />
        </Provider>
      );
    });

    it('should render landing page', () => {
      history.push("/");
      wrapper.update();
      expect(wrapper.find(Landing).length).toEqual(1);  
    })

    it('should not go to login page', () => {
      history.push("/login");
      wrapper.update();
      expect(wrapper.find(LoginPage).length).toEqual(0);  
    })

    it('should not go to register page', () => {
      history.push("/register");
      wrapper.update();
      expect(wrapper.find(RegisterPage).length).toEqual(0);  
    })

    it('should go to setup page', () => {
      history.push('/setup');
      wrapper.update();
      expect(wrapper.find(PlaidLoginPage).length).toEqual(1);
    })

    it('should go to on dashboard', () => {
      history.push('/dashboard');
      wrapper.update();
      expect(wrapper.find(BillOverview).length).toEqual(1);
    })
  })
});
