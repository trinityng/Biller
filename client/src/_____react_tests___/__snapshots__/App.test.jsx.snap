// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main App component should render correctly as connected component 1`] = `ReactWrapper {}`;

exports[`main App component should render correctly in connected component 1`] = `ReactWrapper {}`;
